### ISSUE TITLE
Globbing Exclusion on Windows doesn't work inside a Target

### ISSUE THREAD

**Issue by eerhardt** [Microsoft/msbuild#2631]
This issue looks a lot like #2621.  I'm not sure it is the same exact issue, but logging it here since this doesn't appear to be a regression - it repros with .NET Core SDK 2.0.2.

Here's my repro:

[test.zip](https://github.com/Microsoft/msbuild/files/1377967/test.zip)

```xml
<Project>
    <PropertyGroup>
        <PublishDir>$(MSBuildThisFileDirectory)</PublishDir>
    </PropertyGroup>
    <Target Name="Build">
        <ItemGroup>
            <SdkFiles Include="$(PublishDir)/**/*"
                      Exclude="$(PublishDir)/Sdks/**/*" />
        </ItemGroup>
    </Target>
</Project>
```

Unzip that and run `dotnet.exe msbuild .\test.proj /v:diag` and inspect the `SdkFiles` item.

You'll notice that the `SdkFiles` item contains files from the `Sdks` folder, which should have been excluded.

## Note
1. This only repros on Windows, it doesn't seem to repro on Linux.
1. If I move the `<ItemGroup>` outside of the `<Target>`, then the globbing works as expected - the files in the `Sdks` folder are excluded.
1. One thing that leads me to believe my scenario is different is because I can repro this using both the `2.0.2` .NET Core SDK and the `2.1.0-preview` .NET Core SDK from master.

**Comment by nguerrera** [comment]
@cdmihai

**Comment by cdmihai** [comment]
Closing as duplicate of #2621

**Comment by eerhardt** [comment]
FYI - The reason I didn't think it was duplicated is because it repros with `2.0.2`.  and I just tested it with `2.0.0` and it repros there as well.

Can you ensure when you fix #2621, you fix this repro as well?
